### DISCUSSION TITLE
Major release

### DISCUSSION THREAD

**Discussion by vladmandic** [vladmandic/sdnext#2316] [Announcements]
After baking for couple of weeks, dev branch has been merged to master.

This is a major release, with many changes and new functionality...  

Changelog is massive, but do read through or you'll be missing on some very cool new functionality  
or even free speedups and quality improvements (regardless of which workflows you're using)!  

Note that for this release its recommended to perform a clean install (e.g. fresh `git clone`)  
Upgrades are still possible and supported, but clean install is recommended for best experience  
(primary issue with upgrades is not configuration (you can carry over your `config.json` if desired),  
its handling of `git submodule` which need reinitialization)  

For full details, see: https://github.com/vladmandic/automatic/discussions/99#discussioncomment-7305431

**Comment by danaildichev** [comment]
this plus the news about nvidia's rtx plugin makes me scared to touch anything now. 🙈

**Reply by a2ZxgdqHeSs8F51cIyuK** [reply]
Speaking of the Nvidia plugin, did Vlad say anything anywhere about it? I looked all over but I only saw references to something from months ago.
I tried installing it but it just gives me errors.

**Reply by ghost** [reply]
I submitted a bug report to the devs of the Nvidia plugin and they noted what would be needed to get that working with SD.NEXT. I submitted that information to Vlad here: https://github.com/vladmandic/automatic/issues/2329

**Comment by Chinico68** [comment]
Sorry to say but new GUI is not as good as before I'd prefer the tabs to the arrows for Batch, Resize... etc... Not working properly on Microsoft edge, rendering windows you need to scroll down to the bottom in Firefox too... previous layout was way better... (SD Next is always the best anyway) but new GUI... :(

**Reply by mkossigit** [reply]
Yes find the new tabs frustrating, especially the second pass button being all the way down below instead of up the top.

**Comment by Tillerz** [comment]
Hi,

nice that there's some clean up. It needs some fixes though. :-p

1. "brownian noise" and "discard penultimate sigma" are not recognized for UI changes, so you cannot save their state as default.

2. The quicksettings get wrapped around into the next line at the middle of the screen, if they are too long, not using full screen width. This is causing it: #quicksettings { width: fit-content; }. Adding #quicksettings { width: unset; } to the local css file fixed it for me.

3. There's a little bit of trouble with the left-side menus:

When you click the little arrow to expand, the left menu column gets WIDER to show a really NARROW menu. It really shouldn't expand in width and also the stuff of that 'tab' show use the whole space available, and not try to stay as small as possible:

<img width="954" alt="11c84b5c63b2eb2ec38168c8e70e20da" src="https://github.com/vladmandic/automatic/assets/63370525/385c2918-e5b8-4ad3-8040-ac697f1718cc">

**Reply by vladmandic** [reply]
Make sure you're using default black-teal theme. Other themes will be updated shortly.

**Comment by GalaxyTimeMachine** [comment]
I'm having intermittent problems with images ending with some noise still remaining (see below) after changing from default to a different sampler, even from default to DEIS (which should be the same). If I make other changes (eg. disable and re-enable 2nd pass), it sometimes clears it, but not always. Have to restart the server to clear it if nothing else works.
Using DPM++ 2M...
![00005-SDXL-protovisionXLHighFidelity3D_release0620Bakedvae](https://github.com/vladmandic/automatic/assets/52193044/8f88b561-1337-44d8-b481-5887c0b0d4e1)
I was then trying to switch back to "default" (DEIS) to show how it should look...but it's stuck in this incomplete image rendering.
It then worked with UniPC...
![00006-SDXL-protovisionXLHighFidelity3D_release0620Bakedvae](https://github.com/vladmandic/automatic/assets/52193044/bdd9716e-f0d4-47df-8528-4e558d2d5975)
So, I tried "default" again...
![00008-SDXL-protovisionXLHighFidelity3D_release0620Bakedvae](https://github.com/vladmandic/automatic/assets/52193044/3ebc96ff-688b-4f21-9d0a-3d967c776859)

**Reply by GalaxyTimeMachine** [reply]
Is anyone else seeing this problem? I'd like some confirmation that it's not a local issue before opening a bug.

**Reply by MaxJa4** [reply]
Looks similar for me too using DPM++ 2M (karras, low order). DEIS looks better but slightly noisy still. The other ones like UniPC look fine, though seemingly more noisy than SD1.5 images (may be a model issue though).

**Reply by anothertal3** [reply]
~I'm also having issues when trying to select "DPM++ 2M" and "karras" (well, anything, really).~

~Nevermind, I don't know what's happening here. Tested it a couple of times and would always receive pictures full of artifacts. Came here to confirm this. Tried again afterwards and it seems to work now... what the...~

And it's back. Getting artifacts during live preview and randomly in output.

**Reply by GalaxyTimeMachine** [reply]
Opened https://github.com/vladmandic/automatic/issues/2351

**Reply by Tillerz** [reply]
I tried a lora and the whole image had lots of artefacts. Maybe try if you have a lora that messes up your output.

**Comment by Chinico68** [comment]
Any idea why my rendering window is at the very bottom of the page?? black teal or not I get always a huge horizontal scrolling bar as well..
![SCR-20231018-njqs](https://github.com/vladmandic/automatic/assets/21184744/8574a149-1d75-48f6-8012-984cbad67201)

**Reply by Chinico68** [reply]
Got it... my fault :)

**Comment by gsgoldma** [comment]
The new update has broken the civitai helper extension again, the forked one that worked no longer works.

**Comment by Chinico68** [comment]
Hi there me again... is it only me or... there's no Save button for Styles?? Can't see it anywhere :(

**Reply by vladmandic** [reply]
networks -> toolbar on the top right.

**Comment by ghost** [comment]
Hi, I am wondering if anyone else has found a potential issue with the FreeU implementation?
Before I open a bug report, just wanted to know if anyone else has played with it and see what I see.
I created a very large grid of 196 images. The file is 128MB so I had to add to WeTransfer (link good for 1 week):
https://we.tl/t-Ajwz4TBL4q

**I don't see any difference at all between any of the 196 photos, with B settings in FreeU set between 0 and 3 at half steps.** 

Is anyone else experiencing issues with FreeU not changing the images per settings?

**Reply by vladmandic** [reply]
which commit? there was a freeu update yesterday that should fix exactly that.

**Reply by ghost** [reply]
08:53:03-832567 INFO     Version: app=sd.next updated=2023-10-19 hash=632fab30
                         url=https://github.com/vladmandic/automatic.git/tree/master

**Comment by vladmandic** [comment]
i just tried and i can see significant differences with free-u enabled:
![image](https://github.com/vladmandic/automatic/assets/57876960/ad3fd09d-23ac-4db3-85e9-fd3f16973268)

**Reply by ghost** [reply]
I git pulled again and am seeing different results with FreeU in SDXL. No current issue. Thank you!

**Reply by Tillerz** [reply]
The release notes say it's disabled for SDXL currently? I see no difference ( 53d0cb35 ).

**Reply by ghost** [reply]
Maybe I don't know if I'm doing it right, but I generated XYZ grids with different FreeU settings and got different images using SDXL. That wasn't the case the other day. I honestly don't know if it's the FreeU settings causing the difference, but it appeared to work for me. Maybe do another git pull to update your local installation?

**Comment by Tillerz** [comment]
Bugs?

- The separation of "KARRAS" etc into their own setting ... this has the effect that if you load the config from the META of an older image, you have to adjust all the things manually because it doesn't find the renderers/scalers anymore because the prefix for scalers or the postfix for renderers is missing.

- ~~Loading old configs: I cannot reproduce ANY old image anymore. I checked all the settings on txt2img tab, nothing I can switch to get old images rendered again. :-(~~ Ok, fixed. Apparently 1. "brownian noise" needs to be enabled for old metadata, 2. FreeU needs to be off.

- Just a thing I saw: when you are in the SD config and have clicked "Show all pages" and then edit something, like toggling a checkbox, the page always scrolls/jumps to the top, that's not... helpful. :D #2367

Weird:

- And I have a question about gradio... it looks like it creates a LOT of stuff in x:/temp/gradio. I had like 2 GB of stuff in there. I deleted it. But after a SD.next restart it then complained and I had to manually recreate x:/temp/gradio folder so SD can write its stuff. 1. is there no cleanup ever on this folder? 2. should sd.next maybe create that folder automatically?

- "Show MOTD" ... is that actually doing anything? It's always empty for me (or rather: N/A). :)

**Reply by ghost** [reply]
I confirm I have that exact same "Show All Pages" behavior where it jumps to the top after any edit. Thanks for raising that.

**Reply by vladmandic** [reply]
i can confirm that issue, but note that multiple comments burried in different discussion threads are not likely going to get fixed - please create issue if it needs fixing, otherwise there is no chance to follow whats going on.

**Comment by cyatarow** [comment]
Question about HyperTile

According to [the description of the original repo](https://github.com/tfernd/HyperTile#interested-in-integrating-it-into-your-preferred-web-ui), there are two parameters, `vae_chunk` and `unet_chunk`, but the setting screen only has the one for UNet. Where is the one for VAE?

![image](https://github.com/vladmandic/automatic/assets/26016252/dd8417fa-85d4-4971-9748-dd42a67304ee)

**Reply by vladmandic** [reply]
in my tests vae_chunk did not contribute much to performance and it did introduce a lot of complexities regarding compatibility of hypertile in different scenarios (basically, in anything but simplest txt2img without hires or refine or any extensions). currently its implemented and hidden. i don't want to have 100 different questions and issues "why am i getting runtime errors".

**Reply by cyatarow** [reply]
I see, vae_chunk doesn't seem to be very effective.

**Comment by yoshi244** [comment]
Did the formatting to saving images to a specified directory get changed today? Yesterday it always saved the images to the directory I specified it which is not the same directory as the SD.Next folder but the same drive, but now it forces to create the same directory structure within where I have SD.Next at.

**Reply by mkossigit** [reply]
Same issue here.. issue was raised and fixed but still doing it after updating.
Also noticing different saving behaviour.. no sequence number adding on filename. Low res img saves but doing hires after overwrites it. Live preview not updating either

**Comment by FugueSegue** [comment]
![Untitled-1](https://github.com/vladmandic/automatic/assets/59415458/067d5e06-8f22-4f08-a719-600a0ec0a500)

Under the From Image tab in the Inpaint tab, the Mask mode, Masked content, and Inpaint area are stacked one on top of the other and seems to waste space. Is there something I'm doing wrong? Is this normal?

**Reply by vladmandic** [reply]
not intentional, will be fixed today.

**Comment by Chinico68** [comment]
New Issue..
Image grid shows 8 different images, but the generated images are only 4 and all the same... as you can see from the thumbnails... ? Any idea?
![sameoutput](https://github.com/vladmandic/automatic/assets/21184744/196e1436-ebad-4009-9cb6-92901ac431de)

**Reply by Chinico68** [reply]
Now they are 8 in the grid but the first one is showing up 8 times and is the first one I rendered as a try, not even present in the actual image grid... what's wrong? did a git pull this morning... (full clean install for the new Master, running debug almost every time... no errors)
![onlyfirstoutput](https://github.com/vladmandic/automatic/assets/21184744/b93c8e70-39ee-4d5c-87a0-9d22e8cbe544)

**Reply by Chinico68** [reply]
I think the problem is the Filename sanitizer... I think it keeps changing all the images file name to the same one... that's why I have only one file in the directory....

**Reply by Chinico68** [reply]
OK sorted out I just went back to a 2 days old images.py (thank you timeMachine) and now is working fine. ;)

**Reply by vladmandic** [reply]
will be fixed today.

**Comment by Chinico68** [comment]
Me again I am sorry but this is all too much annoying and frustrating. menus or bars are invisible, like images loaded in Controlnet, looks like they are out fo some invisible frame, in latent couple some menus are somewhere hidden if I chosoe a 712x512... I tried  user.css as well but nothing to do, looks like the frame on the left is not dynamic... god know I am going crazy on that thing :(
![lost](https://github.com/vladmandic/automatic/assets/21184744/8de5d59c-9cf9-4c61-8731-ccc03614c206)

**Reply by Tillerz** [reply]
Switch to the default theme for now. Not all themes are up to date to the gradio changes done: Settings / User Interface: black-teal

**Reply by vladmandic** [reply]
@Chinico68 commenting on a totally unrelated thread does help at all - please take 3 min to create an issue and document it fully so it can be reproduced and hopefully fixed.

**Comment by YukiSakuma** [comment]
How do you disable the live preview in this new version? That option seemed to have vanished, I need to disable it for the speedup

**Reply by vladmandic** [reply]
Set previews to 0 or -1. Thar always worked, second option was just redundant.
